src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        linkLastElement(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            linkLastElement(value);
+        } else {
+            linkBeforeElement(value, searchForNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list.isEmpty()) {
+            return;
+        }
+        Node<T> previous = last;
+        T[] elements = (T[]) list.toArray();
```

Comment:
Casting the result of `list.toArray()` to `(T[])` can lead to a `ClassCastException` at runtime if the list is not of the expected type. Consider using a safer approach to convert the list to an array of type `T`.